File: android/app/src/main/kotlin/com/github/jing332/alistflutter/bridge/AppConfigBridge.kt
package com.github.jing332.alistflutter.bridge

import com.github.jing332.alistflutter.config.AppConfig
import com.github.jing332.pigeon.GeneratedApi


object AppConfigBridge : GeneratedApi.AppConfig {
    override fun isWakeLockEnabled() = AppConfig.isWakeLockEnabled

    override fun isStartAtBootEnabled() = AppConfig.isStartAtBootEnabled

    override fun isAutoCheckUpdateEnabled() = AppConfig.isAutoCheckUpdateEnabled
    override fun isAutoOpenWebPageEnabled() = AppConfig.isAutoOpenWebPageEnabled
    override fun getDataDir() = AppConfig.dataDir

    override fun setDataDir(dir: String) {
        AppConfig.dataDir = dir
    }

    override fun isSilentJumpAppEnabled(): Boolean = AppConfig.isSilentJumpAppEnabled

    override fun setSilentJumpAppEnabled(enabled: Boolean) {
        AppConfig.isSilentJumpAppEnabled = enabled
    }

    override fun setAutoOpenWebPageEnabled(enabled: Boolean) {
        AppConfig.isAutoOpenWebPageEnabled = enabled
    }

    override fun setAutoCheckUpdateEnabled(enabled: Boolean) {
        AppConfig.isAutoCheckUpdateEnabled = enabled
    }

    override fun setStartAtBootEnabled(enabled: Boolean) {
        AppConfig.isStartAtBootEnabled = enabled
    }

    override fun setWakeLockEnabled(enabled: Boolean) {
        AppConfig.isWakeLockEnabled = enabled
    }
}

File: ios/Runner/Alist/bridge/AppConfigBridge.swift
//
//  AppConfigBridge.swift
//  Runner
//
//  Created by apple on 2024/8/15.
//

import Foundation
import Alistlib

/**
 * Alist 一些参数配置, 这些应该在 Flutter 实现
 */
class AppConfigBridge:NSObject, AppConfig {
    
    static var instance = AppConfigBridge()
    
    private override init() {}

    func isWakeLockEnabled() throws -> Bool {
        let retVal = UserDefaults.standard.bool(forKey: "WakeLockEnabled")
        if(retVal){
            return retVal
        } else {
            return false
        }
    }

    func isStartAtBootEnabled() throws -> Bool {
        let retVal = UserDefaults.standard.bool(forKey: "StartAtBootEnabled")
        if(retVal){
            return retVal
        } else {
            return false
        }
    }

    func isAutoCheckUpdateEnabled() throws -> Bool {
        let retVal = UserDefaults.standard.bool(forKey: "AutoCheckUpdateEnabled")
        if(retVal){
            return retVal
        } else {
            return false
        }
    }

    func isAutoOpenWebPageEnabled() throws -> Bool {
        let retVal = UserDefaults.standard.bool(forKey: "AutoOpenWebPageEnabled")
        if(retVal){
            return retVal
        } else {
            return false
        }
    }

    func getDataDir() throws -> String {
        let dataDir = UserDefaults.standard.string(forKey: "DataDir")
        if (dataDir != nil && !dataDir!.isEmpty){
            return dataDir!
        } else {
            let documentsDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
            return documentsDirectory.path
        }
    }

    func setDataDir(dir: String) throws {
        UserDefaults.standard.set(dir, forKey: "DataDir")
    }

    func isSilentJumpAppEnabled() throws -> Bool {
        let retVal = UserDefaults.standard.bool(forKey: "SilentJumpAppEnabled")
        if(retVal){
            return retVal
        } else {
            return false
        }
    }

    func setSilentJumpAppEnabled(enabled: Bool) throws {
        UserDefaults.standard.set(enabled, forKey: "SilentJumpAppEnabled")
    }

    func setAutoOpenWebPageEnabled(enabled: Bool) throws {
        UserDefaults.standard.set(enabled, forKey: "AutoOpenWebPageEnabled")
    }

    func setAutoCheckUpdateEnabled(enabled: Bool) throws {
        UserDefaults.standard.set(enabled, forKey: "AutoCheckUpdateEnabled")
    }

    func setStartAtBootEnabled(enabled: Bool) throws {
        UserDefaults.standard.set(enabled, forKey: "StartAtBootEnabled")
    }

    func setWakeLockEnabled(enabled: Bool) throws {
        UserDefaults.standard.set(enabled, forKey: "WakeLockEnabled")
    }
}
